src/lib/comms/sol-coap.c

```diff
@@ -2010,13 +2022,14 @@ sol_coap_unobserve_by_token(struct sol_coap_server *server, const struct sol_net
         if (!match_observe_reply(reply, token, tkl))
             continue;
 
-        reply->cb((void *)reply->data, server, NULL, NULL);
-        sol_ptr_vector_del(&server->pending, i);
```

Comment:
... and again, the unobserve packet will henceforth be sent while the pending_reply is still in the vector. Hope that's OK.